django/core/management/__init__.py

```diff
@@ -382,7 +382,9 @@ def execute(self):
             settings.INSTALLED_APPS
         except ImproperlyConfigured as exc:
             self.settings_exception = exc
-        except ImportError as exc:
+            # The following commands can be run without a valid settings file configured
+            if subcommand not in {"startproject", "startapp", "makemessages"}:
+                sys.stderr.write(str(exc) + "\n")
```

Comment:
https://github.com/django/django/pull/14622#discussion_r797244647 is not addressed.

Also, we now set `self.settings_exception = exc` twice.

Reply:
@felixxm thanks for the review. I made the other changes you suggested, and rebased the branch. Do you have suggestions for how to not write the exception message for these three subcommands, without hard-coding them in the set?

Reply:
How about a boolean attribute on `BaseCommand` akin to `requires_migrations_checks` like `requires_settings`?

Reply:
Thanks. It looks like `subcommand` is still just a string, though, so it might not be able to check a boolean attribute on `BaseCommand`?

Reply:
You have the command class by the end of the function, where you could assign the result of fetch_command(). Is there a way to delay the writing to stderr until that is done?

Reply:
I pushed up a commit adding the `requires_settings` attribute... is that the direction you were thinking?

Reply:
@jacobtylerwalls any thoughts on my latest commit here?